go-client/download_test.go

```diff
@@ -0,0 +1,49 @@
+package exasol_test_setup_abstraction_go
+
+import (
+	"os"
+	"path"
+	"testing"
+)
+
+func TestDownloadFails(t *testing.T) {
+	dir := t.TempDir()
+	localPath := path.Join(dir, "file")
+	var tests = []struct {
+		name          string
+		url           string
+		expectedError string
+	}{
+		{"invalid scheme", "invalid", "download failed: Get \"invalid\": unsupported protocol scheme \"\""},
+		{"connection refused", "http://0.0.0.0/", "download failed: Get \"http://0.0.0.0/\": dial tcp 0.0.0.0:80: connect: connection refused"},
+		{"file not found", "https://example.com/non-existing-file", "download of \"https://example.com/non-existing-file\" failed with status 404: \"404 Not Found\""},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			err := downloadFile(test.url, localPath)
+			if err == nil {
+				t.Error("expected download to fail")
+			} else if err.Error() != test.expectedError {
+				t.Errorf("expected error %q but got %q", test.expectedError, err.Error())
+			}
+		})
+	}
+}
+
+func TestDownloadSucceeds(t *testing.T) {
+	dir := t.TempDir()
+	localPath := path.Join(dir, "file")
+	err := downloadFile("https://example.com", localPath)
```

Comment:
Relying on a 3rd-party website in a test. :thinking:

Reply:
Better than having no test at all. This is a task for future me.